internal/service/certificatesigningrequest.go

```diff
@@ -56,9 +57,16 @@ func signApprovedCertificateSigningRequest(ca *crypto.CA, request api.Certificat
 
 // (DELETE /api/v1/certificatesigningrequests)
 func (h *ServiceHandler) DeleteCertificateSigningRequests(ctx context.Context, request server.DeleteCertificateSigningRequestsRequestObject) (server.DeleteCertificateSigningRequestsResponseObject, error) {
+	allowed, err := auth.GetAuthZ().CheckPermission(ctx, "certificatessigningrequests", "deletecollection")
+	if err != nil {
+		return server.DeleteCertificateSigningRequests401JSONResponse{Message: fmt.Sprintf("auth failed: %v", err)}, nil
+	}
```

Comment:
401 would be "Unauthorized", but when we get to this point in the code, middleware should already have rejected unauthorized requests, or?
Also, we shouldn't return detailed error messages to users, as they can accidentally leak info (I know we still do that in many places in our APIs).

Reply:
Do you think it should be removed? I don't think so

Reply:
No. What I'm wondering is whether a 401 is the appropriate response, because - despite its name being "Unauthorized" it is about attempt to access a resource without proper auth_entication_. Thus, the user would be asked to log in once more, but that wouldn't help.

Would it be a 403 - Forbidden, considering that the effect is the permissions check failed? I also don't think so, because CheckPermissions() failing isn't about lack of privileges, but about an error that's occurring on the side of the service. So... would a 5xx code be more appropriate?

Reply:
500 sounds OK - it sounds like internal server error.